app/queues/dns/create-record-worker.server.ts

```diff
@@ -0,0 +1,148 @@
+import { UnrecoverableError, Worker } from 'bullmq';
+import { RecordStatus } from '@prisma/client';
+
+import { redis } from '~/lib/redis.server';
+import { prisma } from '~/db.server';
+import logger from '~/lib/logger.server';
+import { setMonthsFromNow } from '~/lib/domains.server';
+import { createRecord, getChangeStatus } from '~/lib/dns.server';
+
+import type { Record } from '@prisma/client';
+
+export const addDbRecordQueueName = 'add-db-record';
+export const createDnsRecordQueueName = 'create-dns-record';
+export const checkDnsStatusQueueName = 'check-dns-status';
+export const syncDbStatusQueueName = 'sync-db-status';
+
+export interface JobRecord {
+  username: Record['username'];
+  type: Record['type'];
+  name: Record['name'];
+  value: Record['value'];
+}
+
+export type addDbRecordJobRecordResult = Record['id'];
+export type addDnsRecordJobRecordResult = string;
+
+export const addDbRecordWorker = new Worker<JobRecord, addDbRecordJobRecordResult>(
+  addDbRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in DB');
```

Comment:
All these worker logs should be `.debug()` vs. `.info()`: this info is great to have in development and testing, but not needed in production.